src/pages/workspaces/workspace/analysis/modals/compute-modal-components.ts

```diff
@@ -0,0 +1,71 @@
+import { div, h, label } from 'react-hyperscript-helpers'
+import { Link, Select } from 'src/components/common'
+import { NumberInput } from 'src/components/input'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { useUniqueId } from 'src/libs/react-utils'
+import { computeStyles } from 'src/pages/workspaces/workspace/analysis/modals/modalStyles'
+import { pdTypes } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+export const PersistentDiskType = ({ diskExists, computeConfig, updateComputeConfig }) => {
+  const persistentDiskId = useUniqueId()
+  return (
+    h(div, [
+      label({ htmlFor: persistentDiskId, style: computeStyles.label }, ['Disk Type']),
+      div({ style: { marginTop: '0.5rem' } }, [
+        h(Select, {
+          id: persistentDiskId,
+          value: computeConfig.selectedPersistentDiskType,
+          isDisabled: diskExists,
+          onChange: ({ value }) => updateComputeConfig('selectedPersistentDiskType', value),
```

Comment:
This doesn't seem right... passing a function to update state to a child component feels very wrong... @msilva-broad do you have any insight?